src/charm.py

```diff
@@ -38,21 +33,26 @@ def __init__(self, *args: typing.Any) -> None:
             args: passthrough to CharmBase.
         """
         super().__init__(*args)
-        self._databases = Databases(charm=self)
         self._secret_storage = SecretStorage(charm=self)
-
+        self._database_requirers = Databases.make_database_requirers(self)
         try:
             self._charm_state = CharmState.from_charm(
-                charm=self, secret_storage=self._secret_storage
+                charm=self,
+                secret_storage=self._secret_storage,
+                database_uris=Databases.get_uris(self._database_requirers),
```

Comment:
Similar to previous comment, should this just be a top level function in the module?

Reply:
Updated, thanks